### PR TITLE
Driver/ami430 current range

### LINKED ISSUE #967: driver/AMI430 broken by PR #954
PR #954 breaks the ami @spauka @sohailc .
on qc 0.1.9:
```
ix = AMI430("x", address="192.168.15.121", port=7180)
ix.field(0)
```
leads to:
```
Traceback (most recent call last):

  File "<ipython-input-3-1db96f862287>", line 1, in <module>
    ix.field(0)

  File "c:\qcodes\qcodes\instrument\parameter.py", line 250, in __call__
    self.set(*args, **kwargs)

  File "c:\qcodes\qcodes\instrument\parameter.py", line 417, in set_wrapper
    raise e

  File "c:\qcodes\qcodes\instrument\parameter.py", line 399, in set_wrapper
    set_function(parsed_scaled_mapped_value, **kwargs)

  File "c:\qcodes\qcodes\utils\command.py", line 177, in __call__
    return self.exec_function(*args)

  File "c:\qcodes\qcodes\instrument_drivers\american_magnetics\AMI430.py", line 268, in set_field
    if np.abs(value) > self.field_rating():

  File "c:\qcodes\qcodes\instrument\parameter.py", line 244, in __call__
    return self.get()

  File "c:\qcodes\qcodes\instrument\parameter.py", line 351, in get_wrapper
    raise e

  File "c:\qcodes\qcodes\instrument\parameter.py", line 321, in get_wrapper
    value = get_function(*args, **kwargs)

  File "c:\qcodes\qcodes\utils\command.py", line 177, in __call__
    return self.exec_function(*args)

  File "c:\qcodes\qcodes\instrument_drivers\american_magnetics\AMI430.py", line 167, in <lambda>
    get_cmd=lambda: self.current_rating(),

  File "c:\qcodes\qcodes\instrument\parameter.py", line 244, in __call__
    return self.get()

  File "c:\qcodes\qcodes\instrument\parameter.py", line 351, in get_wrapper
    raise e

  File "c:\qcodes\qcodes\instrument\parameter.py", line 321, in get_wrapper
    value = get_function(*args, **kwargs)

  File "c:\qcodes\qcodes\utils\command.py", line 177, in __call__
    return self.exec_function(*args)

  File "c:\qcodes\qcodes\utils\command.py", line 126, in call_by_str
    return self.exec_str(self.cmd_str.format(*args))

  File "c:\qcodes\qcodes\instrument\base.py", line 648, in ask
    raise e

  File "c:\qcodes\qcodes\instrument\base.py", line 641, in ask
    answer = self.ask_raw(cmd)

  File "c:\qcodes\qcodes\instrument\ip.py", line 181, in ask_raw
    return self._recv()

  File "c:\qcodes\qcodes\instrument\ip.py", line 145, in _recv
    result = self._socket.recv(self._buffer_size)

timeout: ('timed out', "asking 'CURR:RATING?' to <AMI430: x>", 'getting x_current_rating', 'getting x_field_rating', 'setting x_field to 0')
```

### PR DESCRIPTION
**Pull request by WilliamHPNielsen**
Fixes #967 by using `current_limit` instead of `current_rating` for checks. I think this might anyway be a better option.

@spauka and @sohailc, it would be nice if you could try this out on a real instrument (and please report your firmware revision!).

I'll try it out here in Copenhagen to see if my bold fixing claim is even true.

@Dominik-Vogel

### CODE REVIEW

**Comment by WilliamHPNielsen** [comment]
Just tried this out at Copenhagen QT4. Seems to work as it should. The tests pass too (the AV fail is unrelated and happens after the test suite runs), so I'd say: 💃 
@QCoDeS/core?

File: qcodes/instrument_drivers/american_magnetics/AMI430.py [hunk 1]
```diff
@@ -130,79 +140,96 @@ def __init__(self, name, address=None, port=None,
...
+                               get_cmd=lambda: self.current_rating(),
+                               set_cmd=lambda x: self.current_rating(x),
+                               scale=1/float(self.ask("COIL?")))
+
+        self.add_parameter('current_limit',
+                           unit="A",
+                           set_cmd="CONF:CURR:LIMIT {}",
+                           get_cmd='CURR:LIMIT?',
+                           get_parser=float,
+                           vals=Numbers(0, 80))  # what are good numbers here?
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/american_magnetics/AMI430.py]: Should we add field_limit same as field_rating ?

Comment (WilliamHPNielsen) [reply qcodes/instrument_drivers/american_magnetics/AMI430.py]: Ah, yes, of course. My bad.

**Comment by jenshnielsen** [comment]
Appveyor failing due to #1057  not related

**Pull request merged**